### PR TITLE
Fix both document and code defectives

### PR DESCRIPTION
**Pull request by PeiweiHu**

### CODE REVIEW

File: crypto/evp/evp_enc.c [hunk 1]
```diff
@@ -1031,7 +1031,7 @@ int EVP_CIPHER_CTX_set_padding(EVP_CIPHER_CTX *ctx, int pad)
     params[0] = OSSL_PARAM_construct_uint(OSSL_CIPHER_PARAM_PADDING, &pd);
     ok = evp_do_ciph_ctx_setparams(ctx->cipher, ctx->algctx, params);
 
-    return ok != 0;
+    return ok > 0;
```
Comment (mattcaswell) [inline crypto/evp/evp_enc.c:1034]: Looks like there are quite a few similar instances of this bug. I'm looking at: EVP_CIPHER_CTX_original_iv, EVP_CIPHER_CTX_iv, EVP_CIPHER_CTX_iv_noconst, EVP_CIPHER_CTX_get_num, EVP_CIPHER_CTX_get_key_length, EVP_CIPHER_CTX_set_num

**Review by mattcaswell — APPROVED** [review]
Approved in as far as it goes. It would be nice if this PR was extended (or another PR raised) to address the other similar instances.

File: crypto/evp/evp_enc.c (hunk 1 above)
Comment (PeiweiHu) [reply crypto/evp/evp_enc.c:1034]: With pleasure to do that. But I am confused while reviewing that where is the implementation or definition of functions like evp_do_ciph_ctx_setparams. I concluded just by the comment in the header file and usage in other places.

Comment (mattcaswell) [reply crypto/evp/evp_enc.c:1034]: They are implemented here:

https://github.com/openssl/openssl/blob/2c2724476ef50b8926b033f009bdfc85ac3f1816/crypto/evp/evp_utils.c#L20-L81

Notably all of these functions can return EVP_CTRL_RET_UNSUPPORTED (which is defined as -1).

**Review by paulidale — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
The CI failure is relevant and quite weird.

**Comment by t8m** [comment]
Needs rebasing.

**Comment by PeiweiHu** [comment]
Sorry for late response, reopen after rebase and analysis of failure.

**Pull request closed without merging**